Sources/VLCHTTPConnection.m

```diff
@@ -142,31 +142,20 @@ - (BOOL)fileIsInDocumentFolder:(NSString*)filepath
 
 - (NSObject<HTTPResponse> *)_httpGETThumbnailForPath:(NSString *)path
 {
-    NSString *filePath = [[path stringByReplacingOccurrencesOfString:@"/thumbnail/" withString:@""] stringByAddingPercentEncodingWithAllowedCharacters:NSCharacterSet.URLFragmentAllowedCharacterSet];
-    filePath = [filePath stringByReplacingOccurrencesOfString:@".png" withString:@""];
-
-    NSManagedObjectContext *moc = [[MLMediaLibrary sharedMediaLibrary] managedObjectContext];
-    if (moc) {
-        NSPersistentStoreCoordinator *psc = [moc persistentStoreCoordinator];
-        if (psc) {
-            NSManagedObject *mo = nil;
-            @try {
-                mo = [moc existingObjectWithID:[psc managedObjectIDForURIRepresentation:[NSURL URLWithString:filePath]] error:nil];
-            }@catch (NSException *exeption) {
-                return [[HTTPErrorResponse alloc] initWithErrorCode:404];
-            }
+    NSString *filePath = [[path stringByReplacingOccurrencesOfString:@"Thumbnail/" withString:@""] stringByAddingPercentEncodingWithAllowedCharacters:NSCharacterSet.URLFragmentAllowedCharacterSet];
 
-            NSData *theData = UIImageJPEGRepresentation([VLCThumbnailsCache thumbnailForManagedObject:mo], .9);
-            NSString *contentType = @"image/jpg";
+    if ([filePath isEqualToString:@"/"]) return [[HTTPErrorResponse alloc] initWithErrorCode:404];
 
-            if (theData) {
-                HTTPDataResponse *dataResponse = [[HTTPDataResponse alloc] initWithData:theData];
-                dataResponse.contentType = contentType;
-                return dataResponse;
-            }
-        }
-    }
-    return [[HTTPErrorResponse alloc] initWithErrorCode:404];
+    UIImage *thumbnail = [UIImage imageWithContentsOfFile:filePath];
+    if (!thumbnail) return [[HTTPErrorResponse alloc] initWithErrorCode:404];
+
+    NSData *theData = UIImageJPEGRepresentation(thumbnail, .9);
+
+    if (!theData) return [[HTTPErrorResponse alloc] initWithErrorCode:404];
+
```

Comment:
I might be wrong but I think the `if (!thumbnail)`  is not necessary since `UIImageJPEGRepresentation` will return nil if the thumbnail is nil.
Therefore, a single check for `theData` should reduce the code and be sufficient. 🎉